server/internal/database/destination.go

```diff
@@ -22,31 +24,103 @@ func NewDestinationORM() *DestinationORM {
 	}
 }
 
+// encryptDestinationConfig encrypts the config field before saving
+func (r *DestinationORM) encryptDestinationConfig(destination *models.Destination) error {
+	if destination.Config != "" {
+		encryptedConfig, err := utils.EncryptConfig(destination.Config)
+		if err != nil {
+			return fmt.Errorf("failed to encrypt destination config: %s", err)
+		}
+		destination.Config = encryptedConfig
+	}
+	return nil
+}
+
+// decryptDestinationConfig decrypts the config field after reading
+func (r *DestinationORM) decryptDestinationConfig(destination *models.Destination) error {
+	if destination.Config != "" {
+		decryptedConfig, err := utils.DecryptConfig(destination.Config)
+		if err != nil {
+			return fmt.Errorf("failed to decrypt destination config: %s", err)
+		}
+		destination.Config = decryptedConfig
+	}
+	return nil
+}
+
+// decryptDestinationSliceConfigs decrypts config fields for a slice of destinations
+func (r *DestinationORM) decryptDestinationSliceConfigs(destinations []*models.Destination) error {
+	for _, dest := range destinations {
+		if err := r.decryptDestinationConfig(dest); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 func (r *DestinationORM) Create(destination *models.Destination) error {
+	// Encrypt config before saving
+	if err := r.encryptDestinationConfig(destination); err != nil {
+		return err
+	}
+
 	_, err := r.ormer.Insert(destination)
 	return err
 }
 
 func (r *DestinationORM) GetAll() ([]*models.Destination, error) {
 	var destinations []*models.Destination
 	_, err := r.ormer.QueryTable(r.TableName).RelatedSel().All(&destinations)
-	return destinations, err
+	if err != nil {
+		return nil, err
+	}
+
+	// Decrypt config after reading
+	if err := r.decryptDestinationSliceConfigs(destinations); err != nil {
+		return nil, err
+	}
+
+	return destinations, nil
```

Comment:
if the error is being handled by calling function 
```suggestion
	 return r.decryptDestinationSliceConfigs(destinations)
```

Reply:
else use proper error